### PR TITLE
Test files

### PR DESCRIPTION
**Pull request by Shivansh070707**

### CODE REVIEW

**Comment by sumit03guha** [comment]
1. The `artifacts` and the `cache` folder should be placed inside the `build` folder.
2. Each time the test cases are executed, the addresses inside the JSON files in the build folder gets modified. This should not happen while executing the test cases.

**Review by sumit03guha — CHANGES_REQUESTED** [review]
1. The `artifacts` and the `cache` folder should be placed inside the `build` folder.
2. Each time the test cases are executed, the addresses inside the JSON files in the build folder gets modified. This should not happen while executing the test cases.

File: scripts/autofarm.deploy.ts [hunk 1]
```diff
@@ -67,9 +79,16 @@ export async function deployAutofarmDiamond(args: any) {
...
+    if (err) console.error(err);
+  });
   fs.writeFileSync(
-    'Build/Diamond.json',
-    JSON.stringify(diamondFacetData, null, 2)
+    `./build/${CHAIN}/${ENV}/${name}/Diamond.json`,
+    JSON.stringify(diamondFacetData, null, 2),
+    (err) => {
+      if (err) console.error(err);
+    }
```
Comment (sumit03guha) [inline scripts/autofarm.deploy.ts]: - The same/similar function lines are being used repetitively in the code in many places.
Create a separate function and call that in each of those places.
- Also put the `fs.mkdir` and `fs.writeFileSync` in a `try-catch` block.
Reference - https://github.com/sumit03guha/Hardhat-BoilerPlate/blob/a2ea154fa5fee230bfc4904245087fc9648ded23/scripts/deploy.ts#L41

**Review by sumit03guha — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1,17 +1,151 @@
...
-}
+# Autofarm-Diamond Project
+
+## Project Description
+
+---
+
+- The DApp (Decentralised Application) was designed with the purpose of optimising DeFi (Decentralised Finance) users yields as they interact with the other DApps in the DeFi space.
+- Reference -(https://autofarm.gitbook.io/autofarm-network/)
+
```
Comment (sumit03guha) [inline README.md:13]: add more details regarding the EIP-2535 standard

File: README.md [hunk 3]
```diff
@@ -1,17 +1,151 @@
...
+  ```shell
+  npx hardhat run .\scripts\stratx2.deploy --network <NETWORK> param1 param2
+  ```
+
+  ```shell
+  npx hardhat run .\scripts\initFunction --network <NETWORK> param
+  ```
+
+  `<NETWORK>` can be `localhost`, `goerli`, `plsTest`, `plsMain`, `mainnet` or any other network that is supported and defined in the `hardhat.config.js`.
```
Comment (sumit03guha) [inline README.md]: it should be `hardhat.config.ts` and not `js`

File: README.md [hunk 4]
```diff
@@ -1,17 +1,151 @@
...
+
+## Smart contracts size
+
+- To generate the sizes of the smart contracts.
+
+  ```shell
+  npx hardhat size-contracts
+  ```
+
+  ![size](./smart-contract-size.png)
```
Comment (sumit03guha) [inline README.md:180]: the picture is not showing up

File: README.md [hunk 5]
```diff
@@ -1,17 +1,151 @@
...
+
+## Code coverage
+
+- To generate the code coverage report for the test cases of the smart contracts.
+
+  ```shell
+  npx hardhat coverage
+  ```
+
+  ![coverage](./code-coverage.png)
```
Comment (sumit03guha) [inline README.md:192]: the picture is not showing up

File: README.md [hunk 6]
```diff
@@ -1,17 +1,151 @@
...
+
+- Deploy the autofarm project smart contracts using either of the two ways mentioned above.
+
+  ```shell
+  npx hardhat run .\scripts\autofarm.deploy --network <NETWORK>  param1 param2
+  ```
+
+  ```shell
+  npx hardhat run .\scripts\stratx2.deploy --network <NETWORK> param1 param2
+  ```
```
Comment (sumit03guha) [inline README.md]: what are these `param1` and `param2`?

**Review by sumit03guha — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (ShivanshRapidinnovation) [reply README.md]: arguments to pass for running the function

Comment (sumit03guha) [reply README.md]: mention in detail in the readme what type of argument it can be, with examples.

**Pull request merged**